libdnf/module/ModulePackageMaker.cpp

```diff
@@ -0,0 +1,31 @@
+#include "ModulePackageMaker.hpp"
+#include "modulemd/ModuleMetadata.hpp"
+
+#include <memory>
+
+std::map<Id, std::shared_ptr<ModulePackage>> ModulePackageMaker::fromFile(Pool *pool, HyRepo repo, const std::string &filePath)
+{
+    auto metadata = ModuleMetadata::metadataFromFile(filePath);
+    return createModulePackages(pool, repo, metadata);
+}
+
+std::map<Id, std::shared_ptr<ModulePackage>> ModulePackageMaker::fromString(Pool *pool, HyRepo repo, const std::string &fileContent)
+{
+    auto metadata = ModuleMetadata::metadataFromString(fileContent);
+    return createModulePackages(pool, repo, metadata);
+}
+
+std::map<Id, std::shared_ptr<ModulePackage>> ModulePackageMaker::createModulePackages(Pool *pool, HyRepo repo, const std::vector<std::shared_ptr<ModuleMetadata>> &metadata)
+{
+    std::map<Id, std::shared_ptr<ModulePackage>> modules;
+
+    Repo *solvRepo = repo->libsolv_repo;
+    Repo *clonedRepo = repo_create(pool, solvRepo->name);
+
+    for (auto data : metadata) {
+        auto modulePackage = std::make_shared<ModulePackage>(pool, clonedRepo, data);
+        modules.insert(std::make_pair(modulePackage->getId(), modulePackage));
```

Comment:
What is an advantage of keeping id and modulePackage in map when modulePackage has also an ID inside?

Reply:
Libsolv returns only id, which somehow needs to be mapped to ModulePackage.